### ISSUE TITLE
Inconsistent ENV Variables used

### ISSUE THREAD

**Issue by dombarnes** [fastlane/fastlane#6478]
### New Issue Checklist
- [X] Updated fastlane to the latest version
- [X] I have read the [Contribution Guidelines](https://github.com/fastlane/fastlane/blob/master/CONTRIBUTING.md)
### Issue Description

Inconsistent environment variables are used throughout fastlane leading to duplicate entries for tools.

action `latest_testflight_build_number` expects ITUNESCONNECT_USER
Sigh expects SIGH_USERNAME
Deliver expects DELIVER_USERNAME
action `set_changelog` expect FASTLANE_USERNAME
Credentials Manager expects FASTLANE_USER
Produce expects DELIVER_USER or PRODUCE_USERNAME
watchbuild expects FASTLANE_USER
etc...

**Suggestion 1**
Enforce standard naming of environment variables for per-tool settings

```
{tool_name}_USERNAME
{tool_name}_PASSWORD
```

which override any global settings, for example

```
ITUNESCONNECT_USERNAME
ITUNESCONNECT_PASSWORD (or use credential manager)
MEMBERCENTER_USERNAME
MEMBERCENTER_PASSWORD (or use credential manager)
```

or for using the same credentials for all Apple/iTunes related work

```
FASTLANE_USERNAME
FASTLANE_PASSWORD (or use credential manager)
```

**Suggestion 2**
Create documentation listing all global and tool-specific environment variables.

I welcome some feedback and discussion on this.

**Comment by KrauseFx** [comment]
```
FASTLANE_USERNAME
FASTLANE_PASSWORD
```

That's already the case, give it a try 👍 

I agree that things might not be consistent, but that's not something we can change easily, since we have to stay backwards compatible

**Comment by dombarnes** [comment]
I know I've had FASTLANE_USERNAME set before and it's failed to be picked up by some tools I think.

**Comment by KrauseFx** [comment]
@dombarnes that shouldn't be the case. Can you find out what tools and parameters are affected by it?

**Comment by dombarnes** [comment]
To test this, I added fastlane@domain.co.uk as an ENV knowing its an invalid AppleID and has no saved credentials for it, and it would/should prompt for a password and return an invalid login. In each of these tests, I was prompted for my username instead.
````
latest_testflight_build_number (fastlane core 0.52.1)
sigh (1.11.2)
pilot (1.10.1)
deliver (1.14.2)
spaceship (0.36.0)
````
In my ENV
`FASTLANE_USERNAME=fastlane@domain.com`
If I then set `DELIVER_USERNAME` deliver uses that, same with SIGH_USERNAME, PILOT_USERNAME

FASTLANE_USER seems to work a lot of tools, although the `set_changelog` action seems to request FASTLANE_USERNAME.
